### ISSUE TITLE
ThemeBase should have a Logger and possibly an AddThemeMessage just like the ModuleBase

### ISSUE THREAD

**Issue by vnetonline** [oqtane/oqtane.framework#2976]
The `ThemeBase` should have a `Logger` and possibly an `AddThemeMessage` just like the `ModuleBase` so that theme related errors can be logged and handled using the `AddThemeMessage` or the `Logger` to silently capture errors.

While building the `Amazing.Theme.NiceAdmin` I have realised there is no way of logging an error and handling it nicely.

I could create an `AmazingThemeControlBase which extends from `ThemeControlBase` and initialise a `logger` and `AddThemeMessage` in there but I think when other theme builders start building themes we should have this as part of the Framework.

What are your thoughts @sbwalker & @leigh-pointer ?

**Comment by sbwalker** [comment]
The original concept of themes was that they provide look and feel to your application - not functionality.  This is why most of the example themes are very simple and have no business logic. The advanced functionality in a theme is usually encapsulated in Theme Controls (ie. menu, login, logo, etc...)... or within its Panes where Modules can be injected at run-time. 

That being said, this is an artificial restriction as it is completely possible in Oqtane for developers to build an entire application within a theme. For example, they could choose to ignore the module concept and embed all of their logic within the theme directly. This approach would still allow the custom code to be maintained and packaged separately from the framework. The challenge is that none of the existing Module Controls (ie. FileManager, Section, Panel, etc...) will function in this context as they all currently depend on ModuleState.

A good example is the AddModuleMessage() method in ModuleBase. It depends on ModuleInstance (a parent component) which does not exist for a theme. So a Theme would need its own unique implementation. However where would the theme message actually be displayed? There is no "standard" location in a theme as there is no required container. So I guess it would need to be injected at the top of the page above the theme itself? This is going to look rather strange.

So basically what I am saying is that this needs some more thought before it is added to the framework.

**Comment by sbwalker** [comment]
Yes, the logging methods could be added to ThemeBase. The logic would need to be modified slightly as the logging methods contain references to ModuleState (which does not exist in a theme). ModuleId could be set to null and PageId can be obtained from PageState

**Comment by vnetonline** [comment]
Thanks I was going to send a pull request but you beat me to it.

Exactly what I did looking at your commit
